pkg/graph/schema.graphql

```diff
@@ -2001,6 +2002,41 @@ input SendTRBAdviceLetterInput {
   notifyEuaIds: [String!]!
 }
 
+"""
+Represents a recommendation and links that have been added to a TRB advice letter
+"""
+type TRBAdviceLetterRecommendation {
+  id: UUID!
+  trbRequestId: UUID!
+  title: String!
+  recommendation: String!
+  links: [String!]!
+  createdBy: String!
+  createdAt: Time!
+  modifiedBy: String
+  modifiedAt: Time
+}
+
+"""
+The input required to add a recommendation & links to a TRB advice letter
+"""
+input CreateTRBAdviceLetterRecommendationInput {
+  trbRequestId: UUID!
+  title: String!
+  recommendation: String!
+  links: [String!]!
+}
+
+"""
+The input required to update a recommendation to a TRB advice letter
+"""
+input UpdateTRBAdviceLetterRecommendationInput @goModel(model: "map[string]interface{}") {
+  id: UUID!
```

Comment:
It works both ways, but I'd recommend making `id: UUID!` a different parameter than the `changes` (which gets converted to `map[string]interface{}`, because then you have schema-level validation on the existence and type of `id`, like how we do for `updateTRBRequest`:

```graphql
updateTRBRequest(id: UUID!, changes: TRBRequestChanges): TRBRequest!
```

This makes it so you don't have to do this block of code in the resolver:

```golang
	idStr, idFound := changes["id"]
	if !idFound {
		return nil, errors.New("missing required property trbRequestId")
	}

	id, err := uuid.Parse(idStr.(string))
	if err != nil {
		return nil, err
	}
```

since `id` is required and will automatically be mapped to the right type.

Reply:
For what it's worth, I see we do this both ways in the app, and I'd go with whatever way works best, but wanted to provide my 2c on how it can simplify having to write parameter validation in resolvers!

Reply:
Per a discussion between Jeremy and I, we're not going to change this (fits an existing paradigm within the app to have a single input for mutations!)